test/csuite/incr_backup/main.c

```diff
@@ -0,0 +1,846 @@
+/*-
+ * Public Domain 2014-2020 MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * This program tests incremental backup in a randomized way. The random seed used is reported and
+ * can be used in another run.
+ */
+
+#include "test_util.h"
+
+#include <sys/wait.h>
+#include <signal.h>
+
+#define ITERATIONS 10
+#define MAX_NTABLES 100
+
+#define MAX_KEY_SIZE 100
+#define MAX_VALUE_SIZE 10000
+#define MAX_MODIFY_ENTRIES 10
+#define MAX_MODIFY_DIFF 500
+
+#define URI_MAX_LEN 32
+#define URI_FORMAT "table:t%d-%d"
+#define KEY_FORMAT "key-%d-%d"
+
+static int verbose_level = 0;
+static uint64_t seed = 0;
+
+static void usage(void) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+
+/*
+ * Note: set this to true to copy incremental files completely.
+ */
+static bool slow_incremental = false;
+
+/* TODO: rename and drop are not currently working, they give resource busy. */
+static bool do_rename = false;
+static bool do_drop = false;
```

Comment:
Ravi saw something like this when he tried to do some testing with the example program. He was also seeing EBUSY when I asked him why he was closing and reopening the connection entirely. Is it due to the table being left open or cache cursors on that table or something? If you're sure all cursor to the table are closed, we should investigate it.

Reply:
rename and drop are returning `EBUSY`.  I don't think I've left cursors open, I can investigate further.

Reply:
Don't take too much time on this right now. I'd rather get this committed and open up follow on work for after the release.